### PR TITLE
[Pybind] Check version of CMAKE_CXX_COMPILER_VERSION

### PR DESCRIPTION
**Pull request by mryzhov**
### Details:
Python bindings compiled with the 14.28 toolset (19.28 compiler) could not be imported by another Pybind module linked with OpenVINO

### Tickets:
 - *149694*

### CODE REVIEW

**Review by ilya-lavrenov — APPROVED** [review]

**Comment by ilya-lavrenov** [comment]
Let's merge Jenkins first.

**Comment by mryzhov** [comment]
depends on https://github.com/intel-innersource/frameworks.ai.openvino.ci.product-configs/pull/3700

File: src/bindings/python/CMakeLists.txt [hunk 1]
```diff
@@ -79,7 +79,18 @@ function(ov_check_python_build_conditions)
...
 
-    if((Python3_Development.Module_FOUND OR Python3_Development_FOUND) AND NOT python_debug)
+    # Python bindings compiled with the 14.28 toolset (19.28 compiler) could not be imported by another Pybind module linked with OpenVINO
+    if (MSVC)
+        if(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 19.28 AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 19.29)
+            set(msvc_supported OFF)
+            message(${message_mode} "MSVC toolset version 14.28.x (19.28.x compiler) is not supported. Please update your toolset (${CMAKE_CXX_COMPILER_VERSION}).")
+        else()
+            set(msvc_supported ON)
+            message(STATUS "MSVC toolset version: ${CMAKE_CXX_COMPILER_VERSION}")
```
Comment (mryzhov) [inline src/bindings/python/CMakeLists.txt:89]:
```suggestion
```

File: src/bindings/python/CMakeLists.txt [hunk 2]
```diff
@@ -79,17 +79,16 @@ function(ov_check_python_build_conditions)
         set(python_debug OFF)
     endif()
 
-    # Python bindings compiled with the 14.28 toolset (19.28 compiler) could not be imported by another Pybind module linked with OpenVINO
-    if (MSVC)
+    set(is_compiler_supported ON)
+    if(MSVC)
+        # Python bindings compiled with the 14.28 toolset (19.28 compiler) could not be imported by another Pybind module linked with OpenVINO
         if(CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 19.28 AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 19.29)
```
Comment (ilya-lavrenov) [inline src/bindings/python/CMakeLists.txt:85]:
```suggestion
        if(CMAKE_CXX_COMPILER_ID STEQUAL "MSVC" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 19.28 AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 19.29)
```

File: src/bindings/python/CMakeLists.txt [hunk 3]
```diff
@@ -79,7 +79,14 @@ function(ov_check_python_build_conditions)
         set(python_debug OFF)
     endif()
 
-    if((Python3_Development.Module_FOUND OR Python3_Development_FOUND) AND NOT python_debug)
+    set(is_compiler_supported ON)
+    # Python bindings compiled with the 14.28 toolset (19.28 compiler) could not be imported by another Pybind module linked with OpenVINO
+    if(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 19.28 AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 19.29)
```
Comment (ilya-lavrenov) [inline src/bindings/python/CMakeLists.txt:84]: let's prohibit all versions older than 19.29

**Pull request merged**